python/ray/serve/task_processor.py

```diff
@@ -0,0 +1,488 @@
+            **options,
+        )
+
+        return TaskResult(
+            id=task_response.id,
+            status=task_response.status,
+            created_at=time.time(),
+            result=task_response.result,
+        )
+
+    def get_task_status_sync(self, task_id) -> TaskResult:
+        task_details = self._app.AsyncResult(task_id)
+        return TaskResult(
+            id=task_details.id,
+            result=task_details.result,
+            status=task_details.status,
+        )
+
+    def start_consumer(self, **kwargs):
+        """Starts the Celery worker thread."""
+        if self._worker_thread is not None and self._worker_thread.is_alive():
+            logger.info("Celery worker thread is already running.")
+            return
+
+        self._worker_thread = threading.Thread(
+            target=self._app.worker_main,
+            args=(("worker", f"--hostname={self._app.main}"),),
+        )
+        self._worker_thread.start()
+
+        logger.info(f"Celery worker thread started with hostname: {self._app.main}")
+
+    def stop_consumer(self, timeout: float = 10.0):
+        """Signals the Celery worker to shut down and waits for it to terminate."""
+        if self._worker_thread is None or not self._worker_thread.is_alive():
+            logger.info("Celery worker thread is not running.")
+            return
+
+        logger.info("Sending shutdown signal to Celery worker...")
+
+        # Use the worker's hostname for targeted shutdown
+        self._app.control.broadcast(
+            "shutdown", destination=[f"celery@{self._app.main}"]
+        )
+        self._worker_thread.join(timeout=timeout)
+
+        if self._worker_thread.is_alive():
+            logger.warning(f"Worker thread did not terminate after {timeout} seconds.")
+        else:
+            logger.info("Celery worker thread has stopped.")
+
+        self._worker_thread = None
+
+    def shutdown(self):
+        self._app.control.shutdown()
+
+    def cancel_task_sync(self, task_id) -> bool:
+        return self._app.AsyncResult(task_id).cancel()
+
+    def get_metrics_sync(self) -> Dict[str, Any]:
+        """
+        Returns the metrics of the Celery worker synchronously.
+        More details can be found here: https://docs.celeryq.dev/en/stable/reference/celery.app.control.html#celery.app.control.Inspect.stats
+        """
+        return self._app.control.inspect().stats()
+
+    def health_check_sync(self) -> List[Dict]:
+        """
+        Checks the health of the Celery worker synchronously.
+        Returns a list of dictionaries, each containing the worker name and a dictionary with the health status.
+        Example: [{'celery@192.168.1.100': {'ok': 'pong'}}]
+        More details can be found here: https://docs.celeryq.dev/en/stable/reference/celery.app.control.html#celery.app.control.Control.ping
+        """
+        return self._app.control.ping()
+
+
+def get_task_adapter(config: TaskProcessorConfig) -> TaskProcessorAdapter:
```

Comment:
This is not a public API right?
```suggestion
def _get_task_adapter(config: TaskProcessorConfig) -> TaskProcessorAdapter:
```

Reply:
this is, user will require this in order to enqueue the task, gets its result, etc..
added the publicAPI annotation on the method.